presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloMetadata.java

```diff
@@ -0,0 +1,536 @@
+    {
+        Set<String> schemaNames;
+        if (schemaNameOrNull != null) {
+            schemaNames = ImmutableSet.of(schemaNameOrNull);
+        }
+        else {
+            schemaNames = client.getSchemaNames();
+        }
+
+        ImmutableList.Builder<SchemaTableName> builder = ImmutableList.builder();
+        for (String schemaName : schemaNames) {
+            for (String tableName : client.getTableNames(schemaName)) {
+                builder.add(new SchemaTableName(schemaName, tableName));
+            }
+        }
+        return builder.build();
+    }
+
+    /**
+     * Lists all of the columns for each provided schema table
+     *
+     * @param session Current client session
+     * @param prefix Table prefix
+     * @return Mapping of table names with the given prefix to columns
+     */
+    @Override
+    public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSession session,
+            SchemaTablePrefix prefix)
+    {
+        requireNonNull(prefix, "prefix is null");
+        ImmutableMap.Builder<SchemaTableName, List<ColumnMetadata>> columns =
+                ImmutableMap.builder();
+        for (SchemaTableName tableName : listTables(session, prefix)) {
+            ConnectorTableMetadata tableMetadata = getTableMetadata(tableName);
+            // table can disappear during listing operation
+            if (tableMetadata != null) {
+                columns.put(tableName, tableMetadata.getColumns());
+            }
+        }
+        return columns.build();
+    }
+
+    /**
+     * Gets metadata for the given table
+     *
+     * @param stn Schema and table name
+     * @return Table metadata, or null if schema/table does not exist
+     */
+    private ConnectorTableMetadata getTableMetadata(SchemaTableName stn)
+    {
+        if (!client.getSchemaNames().contains(stn.getSchemaName())) {
+            return null;
+        }
+
+        // Need to validate that SchemaTableName is a table
+        if (!this.listViews(stn.getSchemaName()).contains(stn)) {
+            AccumuloTable table = client.getTable(stn);
+            if (table == null) {
+                return null;
+            }
+
+            return new ConnectorTableMetadata(stn, table.getColumnsMetadata());
+        }
+        else {
+            return null;
+        }
+    }
+
+    /**
+     * List all tables with the given prefix
+     *
+     * @param session Current client session
+     * @param prefix Table prefix
+     * @return List of table names with the given schema and/or table name, which could be one table
+     */
+    private List<SchemaTableName> listTables(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        if (prefix.getSchemaName() == null) {
+            return listTables(session, prefix.getSchemaName());
+        }
+        return ImmutableList.of(new SchemaTableName(prefix.getSchemaName(), prefix.getTableName()));
```

Comment:
This method and others in this class are direct from the example-http connector (which I used to bootstrap the Accumulo connector).  The table returned from this function will eventually get checked for existence.  Regardless, it certainly wouldn't hurt to check here, so I'll do just that.